File: pa1/Lex.java
//Mythili Karra
////mmkarra
////CS101
////04/15/2019
import java.util.Scanner;
import java.io.*;
public class Lex {
	//why throw IOException
	public static void main (String[] args) throws IOException {
	Scanner in = null;
	PrintWriter out = null;
	if(args.length != 2) {
		System.err.println("Usage: FileIO in out");
		System.exit(1);
	}
	int w = 0;
	int l = 0;
	in = new Scanner(new File(args[0]));
	out = new PrintWriter(new FileWriter(args[1]));
	
	while(in.hasNextLine()) {
		l++;
		in.nextLine();	
		}
		in.close();
		in = null;
		in = new Scanner(new File(args[0]));
		out = new PrintWriter(new FileWriter(args[1]));
		String[] array = new String[l];
		while(in.hasNextLine()) {
			array[w] = in.nextLine(); 
			w++;
		}	
		//in.close();
//String[] array = {"one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve","thirteen","fourteen","fifteen","sixteen","seventeen","eighteen","nineteen","twenty"};
	//make a list "a"  with indexes of "array"
//	String[] array= {"one","two","three","four","five"};//"six","seven"};
/*	List a = new List();
	for(int x = 0; x < array.length; x++) {
		a.append(x);
	}*/
	List list = new List();
	int i, j;
	String temp;
//	a.moveFront();
	list.append(0);
	list.moveFront();
	for(j = 1; j < array.length; j++) {
		temp = array[j];
		if(j != 1)
//		a.moveNext();
		while(list.index() >= 0 && temp.compareTo(array[list.get()]) < 0) {
			//a.movePrev();
			list.movePrev();
		}

		if(list.index() < 0) {	
			list.prepend(j);
		}
		else {
			if(list.index() != list.length()-1) {
				list.insertAfter(j);
			}
			else {
				list.append(j);
			}
		}
		 list.moveBack();
//	if(a.index() < 0) {
//		a.moveFront();
//	}
//		while(a.index() != j-1) {
//			a.moveNext();
//		}
	}
	list.moveFront();
	while(list.index() > -1) {
		out.println(array[list.get()]);
		list.moveNext();
	}
	in.close();
	out.close();
}
}

File: pa2/backup/Lex.c
//Mythili Karra
//////mmkarra
//////CS101
//////04/18/2019
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "List.h"
#define MAX_LEN 255
	int  main (int argc, char* argv[]) {
		FILE* in;
		FILE* out;
		if(argc != 3) {
			printf("Usage: FileIO in out");
			exit(1);
		}
		int w = 0;
		int l = 0;
		char x[MAX_LEN];
		in = fopen(argv[1], "r");
		out = fopen(argv[2], "w");
		if(in == NULL) {
			printf("File %s does not exist", argv[1]);
			exit(1);
		}
		if(out == NULL) {
			printf("File %s does not exist", argv[2]);
			exit(1);
		}
		while(fgets(x, MAX_LEN, in) != NULL) {
			l++;
		}
		char array[l][MAX_LEN];
		rewind(in);
		while(fgets(x,MAX_LEN, in)) {
			strcpy(array[w], x);
			w++;
		}
		List list = newList();
		int i, j;
		char* temp;
		append(list,0);
		moveFront(list);
		for(j = 1; j < l; j++) {
			temp = array[j];
			while(index(list) >= 0 && strncmp(temp,array[get(list)],MAX_LEN) < 0) {
				movePrev(list);
			}
			if(index(list) < 0) {
				prepend(list, j);
			}
			else {
				if(index(list) != length(list)-1) {
					insertAfter(list,j);
				}
				else {
					append(list,j);
				}
			}
			moveBack(list);
		}
		moveFront(list);
		while(index(list) > -1) {
			fprintf(out, "%s", array[get(list)]);
			moveNext(list);
		}
		fclose(in);
		fclose(out);
		freeList(&list);
}


